File: PRÁCTICA SEGUNDO PARCIAL/Ejercicio2/Java/EliminarDigitoMenor.java
import java.util.Scanner;

public class EliminarDigitoMenor {
    public static int eliminarDigitoMenor(int numero) {
        String cadena = Integer.toString(numero);
        char digitoMenor = Character.MAX_VALUE;
        for (char digito : cadena.toCharArray()) {
            if (digito < digitoMenor) {
                digitoMenor = digito;
            }
        }
        String nuevaCadena = cadena.replaceFirst(Character.toString(digitoMenor), "");
        return Integer.parseInt(nuevaCadena);
    }

    public static void eliminarDigitoMenorLote() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese los números separados por espacios: ");
        String lote = scanner.nextLine();
        String[] numeros = lote.split(" ");
        for (String numero : numeros) {
            int numeroInt = Integer.parseInt(numero);
            int numeroSinDigito = eliminarDigitoMenor(numeroInt);
            System.out.println("El número " + numero + " sin el dígito menor es: " + numeroSinDigito);
        }
        scanner.close();
    }

    public static void main(String[] args) {
        eliminarDigitoMenorLote();
    }
}

File: PRÁCTICA SEGUNDO PARCIAL/Ejercicio2/Python/EliminarDigitoMenor.py
def eliminar_digito_menor(numero):
    cadena = str(numero)
    digito_menor = min(cadena)
    nueva_cadena = cadena.replace(digito_menor, '', 1)
    if nueva_cadena == '':
        nueva_cadena = '0'
    return int(nueva_cadena)

def eliminar_digito_menor_lote():
    lote = input("Ingrese los números separados por espacios: ")
    numeros = lote.split()
    for numero in numeros:
        numero_sin_digito = eliminar_digito_menor(numero)
        print(f"El número {numero} sin el dígito menor es: {numero_sin_digito}")


eliminar_digito_menor_lote()
